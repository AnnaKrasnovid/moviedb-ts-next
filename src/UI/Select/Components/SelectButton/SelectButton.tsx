import ArrowIcon from '@/UI/ArrowIcon/ArrowIcon';

import { SelectButtonInt } from '@/settings/interfaces';

import styles from './SelectButton.module.scss';

function SelectButton({ isActive, openModal, selectedItem }: SelectButtonInt) {
    return (
        <div className={styles['select-button']} onClick={openModal} data-testid='select-button'>
            <span className={`
                ${styles['select-button__title']} 
                ${styles['select-button__title-button']}`
            }>
                {selectedItem}
            </span>
            <span className={`${styles['arrow']} ${isActive ? styles['arrow_up'] : ''}`} data-testid='arrow'>
                <ArrowIcon isActive={isActive} />
            </span>
        </div>
    )
}

export default SelectButton;