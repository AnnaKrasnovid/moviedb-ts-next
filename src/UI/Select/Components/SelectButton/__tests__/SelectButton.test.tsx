import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import SelectButton from '../SelectButton';

const onClick = jest.fn();

describe('SelectButton', () => {
    it('Рендер компонента по умолчанию', () => {
        render(
            <SelectButton
                isActive={false}
                openModal={onClick}
                selectedItem={'Все жанры'}
            />
        );

        const element = screen.getByText('Все жанры');
        const arrow = screen.getByTestId('arrow');

        expect(element).toBeInTheDocument();
        expect(arrow).not.toHaveClass('arrow_up');
    });

    it('Проверка активного компонента', () => {
        render(
            <SelectButton
                isActive={true}
                openModal={onClick}
                selectedItem={'Все жанры'}
            />
        );

        const element = screen.getByTestId('select-button');
        const arrow = screen.getByTestId('arrow');

        expect(element).toBeInTheDocument();
        expect(arrow).toHaveClass('arrow_up');
    });

    it('Проверка клика по компоненту', async () => {
        render(
            <SelectButton
                isActive={true}
                openModal={onClick}
                selectedItem={'Все жанры'}
            />
        );

        const element = screen.getByTestId('select-button');

        await userEvent.click(element);
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('Snapshot', () => {
        const buttonSelect = render(
            <SelectButton
                isActive={false}
                openModal={onClick}
                selectedItem={'Все жанры'}
            />
        );

        expect(buttonSelect).toMatchSnapshot();
    });
})