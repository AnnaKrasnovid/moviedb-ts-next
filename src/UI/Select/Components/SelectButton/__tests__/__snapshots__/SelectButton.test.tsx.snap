// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectButton Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="select-button"
        data-testid="select-button"
      >
        <span
          class="
                select-button__title 
                select-button__title-button"
        >
          Все жанры
        </span>
        <span
          class="arrow "
          data-testid="arrow"
        >
          <svg
            class="arrow   "
            fill="none"
            height="22"
            viewBox="0 0 25 24"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.5 9L12.5 15L6.5 9"
              stroke="#fff"
              stroke-width="2"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="select-button"
      data-testid="select-button"
    >
      <span
        class="
                select-button__title 
                select-button__title-button"
      >
        Все жанры
      </span>
      <span
        class="arrow "
        data-testid="arrow"
      >
        <svg
          class="arrow   "
          fill="none"
          height="22"
          viewBox="0 0 25 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.5 9L12.5 15L6.5 9"
            stroke="#fff"
            stroke-width="2"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
